v19.1/cost-based-optimizer.md

```diff
@@ -27,38 +27,20 @@ The most important factor in determining the quality of a plan is cardinality (i
+To see whether a query will be run with the cost-based optimizer, run the query with [`EXPLAIN (OPT)`](explain.html#opt-option). The `OPT` option displays a query plan tree, along with some information that was used to plan the query.
 
-For example, the following query (which uses [CockroachDB's TPC-H data set](https://github.com/cockroachdb/cockroach/tree/b1a57102d8e99b301b74c97527c1b8ffd4a4f3f1/pkg/workload/tpch)) returns the query plan tree, which means that it will be run with the cost-based optimizer:
+If the query is unsupported it will return an error message that starts with e.g., `pq: unsupported statement`. In such cases, the query will be run with the legacy heuristic planner. This should be rare since the optimizer [supports most SQL statements](#types-of-statements-supported-by-the-cost-based-optimizer).
 
-{% include copy-clipboard.html %}
-~~~ sql
-> EXPLAIN (OPT) SELECT l_shipmode, avg(l_extendedprice) from lineitem GROUP BY l_shipmode;
-~~~
-
-~~~
-                                     text
-+-----------------------------------------------------------------------------+
-group-by
-├── columns: l_shipmode:15(string!null) avg:17(float)
-├── grouping columns: l_shipmode:15(string!null)
-├── stats: [rows=700, distinct(15)=700]
-├── cost: 1207
-├── key: (15)
-├── fd: (15)-->(17)
-├── prune: (17)
-├── scan lineitem
-│    ├── columns: l_extendedprice:6(float!null) l_shipmode:15(string!null)
-│    ├── stats: [rows=1000, distinct(15)=700]
-│    ├── cost: 1180
-│    └── prune: (6,15)
-└── aggregations [outer=(6)]
-└── avg [type=float, outer=(6)]
-└── variable: l_extendedprice [type=float, outer=(6)]
-(16 rows)
-~~~
+<span class="version-tag">New in v19.1</span>: The default output of `EXPLAIN (OPT)` has been simplified from previous versions. Also, there are several new options that provide more information about how the optimizer is planning a query:
 
-In contrast, queries that are not supported by the cost-based optimizer return errors that begin with the string `pq: unsupported statement: ...` or specific messages like `pq: aggregates with FILTER are not supported yet`. Such queries will use the legacy heuristic planner instead of the cost-based optimizer.
+- [`EXPLAIN (OPT)`](explain.html#opt-option) shows a query plan tree.
+- [`EXPLAIN (OPT, VERBOSE)`](explain.html#opt-verbose-option) shows cost details.
+- [`EXPLAIN (OPT, TYPES)`](explain.html#opt-types-option) shows cost and type details.
+- [`EXPLAIN (OPT, ENV)`](explain.html#opt-env-option) shows all details used by the optimizer to generate the query plan, including:
```

Comment:
I'm similarly mildly against publicizing this feature, since I see its value as being an escape hatch for *us* to get all the information we need from customers about a bad plan, and without that particular use case I don't see what value a customer would get out of it. Like, for instance, I don't think this page really explains why someone would ever want to use it.

Reply:
Also removed.